### PR TITLE
feat(Algebra): Add lemmas {`even`, `odd`}`_iff_iff_mod_two_eq`

### PR DESCRIPTION
**Pull request by IvanRenison**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Nat/Even.lean [hunk 1]
```diff
@@ -94,6 +94,10 @@ theorem add_one_lt_of_even {n m : ℕ} (hn : Even n) (hm : Even m) (hnm : n < m)
   rcases hm with ⟨m, rfl⟩
   omega
 
+theorem even_iff_iff_mod_two_eq : (Even n ↔ Even m) ↔ n % 2 = m % 2 := by
```
Comment (j-loreaux) [inline Mathlib/Algebra/Group/Nat/Even.lean:97]: I was confused by the name. I understood after reading the statement, but I didn't predict it from the name. Please update the PR title also.
```suggestion
theorem even_iff_even_iff_mod_two : (Even n ↔ Even m) ↔ n % 2 = m % 2 := by
```

File: Mathlib/Algebra/Ring/Parity.lean [hunk 2]
```diff
@@ -299,6 +299,10 @@ lemma div_two_mul_two_add_one_of_odd (h : Odd n) : n / 2 * 2 + 1 = n := by
 lemma one_add_div_two_mul_two_of_odd (h : Odd n) : 1 + n / 2 * 2 = n := by
   rw [← odd_iff.mp h, mod_add_div']
 
+lemma odd_iff_iff_mod_two_eq : (Odd n ↔ Odd m) ↔ n % 2 = m % 2 := by
```
Comment (j-loreaux) [inline Mathlib/Algebra/Ring/Parity.lean:302]:
```suggestion
lemma odd_iff_odd_iff_mod_two : (Odd n ↔ Odd m) ↔ n % 2 = m % 2 := by
```

**Comment by IvanRenison** [comment]
I was proving that the chromatic number of odd length and at some point I wanted to do a `rw [← even_iff_iff_mod_two_eq]`. I thought that it may be useful in other cases also, but I can also do it with out this.

**Comment by jcommelin** [comment]
I agree with @j-loreaux here. I suggest that we close this for now. But if it turns out that these lemmas provide genuine qol improvement, then please reopen.

**Comment by IvanRenison** [comment]
Okey, I pull requested what I was making in #20117. Instead of this theorem I used a `show`

**Pull request closed without merging**